R/TaskClassif_melanoma.R

```diff
@@ -0,0 +1,125 @@
+#' @title Melanoma Image classification
+#' @name mlr_tasks_melanoma
+#' @description
+#' Classification of melanoma tumor images.
+#'
+#' The data comes from the 2020 SIIM-ISIC challenge.
+#'
+#' @section Construction:
+#' ```
+#' tsk("melanoma")
+#' ```
+#'
+#' @template task_download
+#'
+#' @source
+#' \url{https://challenge2020.isic-archive.com/}
+#'
+#' @section Properties:
+#' `r rd_info_task_torch("melanoma", missings = FALSE)`
+#'
+#' @references
+#' `r format_bib("melanoma2021")`
+#' task = tsk("melanoma")
+#' task
+NULL
+
+# @param path (`character(1)`)\cr
+#   The cache_dir/datasets/melanoma folder
+constructor_melanoma = function(path) {
+  require_namespaces("curl")
+
+  base_url = "https://huggingface.co/datasets/carsonzhang/ISIC_2020_small/resolve/main/"
+
+  compressed_tarball_file_name = "hf_ISIC_2020_small.tar.gz"
+  compressed_tarball_path = file.path(path, compressed_tarball_file_name)
+  curl::curl_download(paste0(base_url, compressed_tarball_file_name), compressed_tarball_path)
+  utils::untar(compressed_tarball_path, exdir = path)
+  on.exit({file.remove(compressed_tarball_path)}, add = TRUE)
+
+  training_metadata_file_name = "ISIC_2020_Training_GroundTruth_v2.csv"
+  training_metadata = data.table::fread(file.path(path, training_metadata_file_name))
+
+  test_metadata_file_name = "ISIC_2020_Test_Metadata.csv"
+  test_metadata = file.path(path, test_metadata_file_name)
+
+  training_metadata = training_metadata[, split := "train"]
+  test_metadata = setnames(test_metadata,
+    old = c("image", "patient", "anatom_site_general"),
+    new = c("image_name", "patient_id", "anatom_site_general_challenge")
+  )[, split := "test"]
+  metadata = rbind(training_metadata, test_metadata)
```

Comment:
I think `fill = TRUE` was here because we want to fill the response variable of the test data with `NA`, will double-check to confirm

Reply:
ok makes sense, maybe add a comment